tensorflow/core/kernels/map_kernels.h

```diff
@@ -176,6 +173,43 @@ class TensorMapHasKey : public OpKernel {
   }
 };
 
+class TensorMapStackKeys : public OpKernel {
+ public:
+  explicit TensorMapStackKeys(OpKernelConstruction* c) : OpKernel(c) {
+    OP_REQUIRES_OK(c, c->GetAttr("key_dtype", &key_dtype_));
```

Comment:
It's not very ergonomic that the user will have to either iterate through all possible types or know the set of types present in advance to get all keys.

Have you thought of ways to make this more generic?

Reply:
We plan to implement a `TensorMapListKeys` op that returns a `TensorList` containing all the keys. Hence we chose the name `Stack` here.